docs/docs/subscription.fsx

```diff
@@ -23,61 +23,162 @@ We can setup forwarding of those events to be processed by our `update` function
+You're probably wondering why it is a list. This allows you to include dependencies. Later we will use this to make the timer change intervals.
 
-let timer initial =
-    let sub dispatch =
-        JS.setInterval
-            (fun _ ->
-                dispatch (Tick DateTime.Now)
-            )
-            1000
-            |> ignore
-    Cmd.ofSub sub
+Here, the timer subscription is always returned from `subscribe`, so it will stay running as long as the program is running.
+
+Let's look at an example where the timer can be turned off. First we add the field `enabled`.
+*)
+
+module ToggleTimer =
+
+    type Model =
+        {
+            current : DateTime
+            enabled : bool
+        }
 
-Program.mkSimple init update (fun model _ -> printf "%A\n" model)
-|> Program.withSubscription timer
-|> Program.run
+    type Msg =
+        | Tick of DateTime
 
+    let init () =
+        {
+            current = DateTime.MinValue
+            enabled = true
+        }
 
 (**
-In this example program initialization will call our subscriber (once) with inital `Model` state, passing the `dispatch` function to be called whenever an event occurs.
-However, any time you need to issue a message (for example from a callback) you can use `Cmd.ofSub`.
+`update` and `timer` are the same as before.
+*)
+    let update msg model =
+        match msg with
+        | Tick current ->
+            { model with
+                current = current
+            }
+
+    let timer onTick =
+        let start dispatch =
+            let intervalId = 
+                JS.setInterval
+                    (fun _ -> dispatch (onTick DateTime.Now))
+                    1000
+            { new IDisposable with
+                member _.Dispose() = JS.clearInterval intervalId }
+        start
+
+(**
+Next, change the subscribe function to check `enabled` before including the timer subscription.
+*)
+    let subscribe model =
+        [ if model.enabled then
+            ["timer"], timer Tick ]
+
+    Program.mkSimple init update (fun model _ -> printf "%A\n" model)
+    |> Program.withSubscription subscribe
+    |> Program.run
+
+(**
+All that's left is to add user interaction to change `enabled`. The timer will stop or start accordingly.
+
+You probably noticed that `subscribe` returns a list. So let's add another timer that's always on.
+*)
+
+(*** hide ***)
+module ToggleTimer2 =
+
+    open ToggleTimer
+
+(*** show ***)
+    let subscribe model =
+        [
+            if model.enabled then
+                ["user"; "timer"], timer Tick
+            ["page"; "timer"], timer Tick
+        ]
+
+(**
+Subscription IDs need to be unique. It's not necessary to use two ID segments... I just liked it better than camel casing the words.
+
+Small stylistic note: In the timer subscriptions above, I used `let` to define the `start` fn, then return it. You will often see a function return value written like this:
```

Comment:
This feels a bit confuse, because you are referring to a snippet above but then user needs to refer to the snippet below.

I think just referring to the one below is good enough or move the remark above and not have 2 snippets.

Reply:
Do you think I should just leave it out and convert all the subscriptions to return `fun dispatch -> . . .`?

When I first started making my own effects (v3 subs), I saw the anonymous fn syntax used everywhere in Elmish code. But for some reason this just didn't fit in my brain. Using `let f dispatch = . . .; f` made it readable for me, but maybe is just me.

Reply:
(I think it's because Cmd/Sub feels like a data type in its usage, but has a partial-application-based API for its implementation. The asymmetry didn't compute for me.)

Reply:
Could you please show me the 2 versions of the code? I am a bit lost here.

Reply:
Here's the style I used in the docs:

```fsharp
let timer onTick =
    let start dispatch = // name fn
        . . .
    start // return named fn
```

Here's the style Elmish libraries often use:

```fsharp
let timer onTick =
    fun dispatch -> // return anonymous fn
        . . .
```

I found this very confusing. F# functions are curried anyway, so the anonymous fn return is equivalent to making dispatch a regular parameter:

```fsharp
let timer onTick dispatch =
    . . .
```

Eventually I understood that the anonymous fn return is trying to tell the user to only provide the first parameter and a function should be returned. To make the point even more obvious in the docs (and my code), I explicitly define a named `start` function and return it.

But maybe that's just me. Or perhaps it should use the anonymous function return style to be consistent with Elmish libraries' source code.

Reply:
I see, I don't have a real preference.

In general, I used the anonymous still because I was just mimicking what Eugene wrote.

Reply:
I added a note at the very bottom of the subscription page to demonstrate the difference. Otherwise I kept the named fn approach for more explicitness.